test_tools/sigma_verify_rules.py

```diff
@@ -0,0 +1,222 @@
+    for sys_path in sys.path:
+        if sys_path.endswith(dirname):
+            return sys_path
+    return dirname
+
+def verify_rules_file(rule_file_path, sigma_config, sigma_backend):
+    """Verifies a given file path contains a valid sigma rule.
+
+        Args:
+            rule_file_path: the path to the rules.
+            sigma_config: config to use
+            sigma_backend: A sigma.backends instance
+
+        Raises:
+            None
+
+        Returns:
+            true: rule_file_path contains a valid sigma rule
+            false: rule_file_path does not contain a valid sigma rule
+    """
+
+    logger.debug('[sigma] Reading rules from {0:s}'.format(
+        rule_file_path))
+
+    if os.path.isfile(rule_file_path) == False:
+        logger.error('Rule file not found')
+        return False
+
+    path, rule_filename = os.path.split(rule_file_path)
+
+    with codecs.open(rule_file_path, 'r', encoding='utf-8') as rule_file:
+        try:
+            rule_file_content = rule_file.read()
+            parser = sigma_collection.SigmaCollectionParser(
+                rule_file_content, sigma_config, None)
+            parsed_sigma_rules = parser.generate(sigma_backend)
+        except NotImplementedError:
+            logger.error('{0:s} Error with file {1:s}'
+            .format(rule_filename, rule_file_path), exc_info=True)
+            return False
+        except (sigma.parser.exceptions.SigmaParseError, TypeError):
+            logger.error(
+                '{0:s} Error with file {1:s} '
+                'you should not use this rule in Timesketch '
+                .format(rule_filename, rule_file_path), exc_info=True)
+            return False
+
+    return True
+
+
+def run_verifier(rules_path, config_file_path):
+    """Run an sigma parsing test on a dir and returns results from the run.
+
+    Args:
+        rules_path: the path to the rules.
+        config_file_path: the path to a config file that contains mapping data.
+
+    Raises:
+        IOError: if the path to either test or analyzer file does not exist
+                 or if the analyzer module or class cannot be loaded.
+
+    Returns:
+        a tuple of lists:
+            - sigma_verified_rules with rules that can be added
+            - sigma_rules_with_problems with rules that should not be added
+    """
+    if not os.path.isdir(rules_path):
+        raise IOError('Rules not found at path: {0:s}'.format(
+            rules_path))
+    if not os.path.isfile(config_file_path):
+        raise IOError('Config file path not found at path: {0:s}'.format(
+            config_file_path))
+
+    sigma_config_path = config_file_path
+
+    with open(sigma_config_path, 'r') as sigma_config_file:
+        sigma_config_con = sigma_config_file.read()
+    sigma_config = sigma_configuration.SigmaConfiguration(sigma_config_con)
+    sigma_backend = sigma_elasticsearch.\
+        ElasticsearchQuerystringBackend(sigma_config, {})
```

Comment:
is it not possible to split the line with 
```
sigma_backend = sigma_elasticsearch.ElasticsearchQuerystringBackend(
    sigma_config, {})
```

?

otherwise you can rename `sigma_backend` to `backend` if that makes it fit better

Reply:
made both